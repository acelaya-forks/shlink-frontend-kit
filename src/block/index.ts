export * from './Message';
export * from './Result';
export * from './SimpleCard';
