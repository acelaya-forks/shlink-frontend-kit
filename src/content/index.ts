export * from './Details';
export * from './Listbox';
export * from './Table';
