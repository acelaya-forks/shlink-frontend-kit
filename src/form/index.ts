export * from './Checkbox';
export * from './ToggleSwitch';
export * from './InputFormGroup';
export * from './LabeledFormGroup';
export * from './SearchField';
