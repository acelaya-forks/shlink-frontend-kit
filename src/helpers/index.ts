export * from './numbers';
export * from './pagination';
