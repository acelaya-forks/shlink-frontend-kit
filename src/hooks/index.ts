import { useRef, useState } from 'react';
import { v4 as uuid } from 'uuid';

type ToggleResult = [boolean, () => void, () => void, () => void];

export const useToggle = (initialValue = false): ToggleResult => {
  const [flag, setFlag] = useState<boolean>(initialValue);
  return [flag, () => setFlag(!flag), () => setFlag(true), () => setFlag(false)];
};

export const useDomId = (): string => {
  const { current: id } = useRef(`dom-${uuid()}`);
  return id;
};

export const useElementRef = <T>() => useRef<T | null>(null);
