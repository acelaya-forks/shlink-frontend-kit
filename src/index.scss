@import './utils/ResponsiveTable';
@import './theme/theme';
