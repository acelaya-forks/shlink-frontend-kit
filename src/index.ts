export * from './block';
export * from './form';
export * from './hooks';
export * from './navigation';
export * from './ordering';
