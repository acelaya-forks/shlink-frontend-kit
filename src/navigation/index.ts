export * from './DropdownBtn';
export * from './RowDropdownBtn';
export * from './NavPills';
