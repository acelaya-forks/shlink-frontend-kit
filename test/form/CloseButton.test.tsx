import { screen } from '@testing-library/react';
import type { CloseButtonProps } from '../../src';
import { CloseButton } from '../../src';
import { checkAccessibility } from '../__helpers__/accessibility';
import { renderWithEvents } from '../__helpers__/setUpTest';

describe('<CloseButton />', () => {
  const setUp = (props: CloseButtonProps = {}) => renderWithEvents(<CloseButton {...props} />);

  it('passes a11y checks', () => checkAccessibility(setUp()));

  it.each([[undefined], ['Click me'], ['Something else']])('sets provided label', (label) => {
    setUp({ label });
    expect(screen.getByLabelText(label ?? 'Close')).toBeInTheDocument();
  });

  it('invokes onClick when clicked', async () => {
    const onClick = vi.fn();
    const { user } = setUp({ onClick });

    await user.click(screen.getByLabelText('Close'));

    expect(onClick).toHaveBeenCalled();
  });

  it.each([
    { solid: true },
    { solid: false },
  ])('has expected classes', ({ solid }) => {
    setUp({ solid });
    expect(screen.getByRole('button').className).toMatchSnapshot();
  });

  it.each([
    'sm' as const,
    'md' as const,
    'lg' as const,
  ])('has icon with expected size', (size) => {
    setUp({ size });
    expect(screen.getByRole('img', { hidden: true })).toMatchSnapshot();
  });
});
