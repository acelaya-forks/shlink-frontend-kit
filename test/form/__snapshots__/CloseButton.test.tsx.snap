// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<CloseButton /> > has expected classes 1`] = `"rounded-md focus-ring cursor-pointer"`;

exports[`<CloseButton /> > has expected classes 2`] = `"rounded-md focus-ring cursor-pointer opacity-50 highlight:opacity-80 transition-opacity"`;

exports[`<CloseButton /> > has icon with expected size 1`] = `
<svg
  aria-hidden="true"
  class="svg-inline--fa fa-xmark "
  data-icon="xmark"
  data-prefix="fas"
  role="img"
  viewBox="0 0 384 512"
>
  <path
    d="M55.1 73.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L147.2 256 9.9 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192.5 301.3 329.9 438.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.8 256 375.1 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192.5 210.7 55.1 73.4z"
    fill="currentColor"
  />
</svg>
`;

exports[`<CloseButton /> > has icon with expected size 2`] = `
<svg
  aria-hidden="true"
  class="svg-inline--fa fa-xmark fa-lg "
  data-icon="xmark"
  data-prefix="fas"
  role="img"
  viewBox="0 0 384 512"
>
  <path
    d="M55.1 73.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L147.2 256 9.9 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192.5 301.3 329.9 438.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.8 256 375.1 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192.5 210.7 55.1 73.4z"
    fill="currentColor"
  />
</svg>
`;

exports[`<CloseButton /> > has icon with expected size 3`] = `
<svg
  aria-hidden="true"
  class="svg-inline--fa fa-xmark fa-xl "
  data-icon="xmark"
  data-prefix="fas"
  role="img"
  viewBox="0 0 384 512"
>
  <path
    d="M55.1 73.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L147.2 256 9.9 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192.5 301.3 329.9 438.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.8 256 375.1 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192.5 210.7 55.1 73.4z"
    fill="currentColor"
  />
</svg>
`;
